File: src/main/java/com/aquasecurity/plugins/trivy/model/Finding.java
package com.aquasecurity.plugins.trivy.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.nio.file.FileSystems;
import java.util.List;


public class Finding {

    @JsonProperty("rule_id")
    public String RuleID;

    @JsonProperty("long_id")
    public String LongID;

    @JsonProperty("rule_description")
    public String RuleDescription;

    @JsonProperty("rule_provider")
    public String Provider;

    @JsonProperty("rule_service")
    public String Service;

    @JsonProperty("impact")
    public String Impact;

    @JsonProperty("resolution")
    public String Resolution;

    @JsonProperty("links")
    public List<String> Links;

    @JsonProperty("description")
    public String Description;

    @JsonProperty("severity")
    public Severity Severity;

    @JsonProperty("status")
    public int Status;

    @JsonProperty("resource")
    public String Resource;

    @JsonProperty("location")
    public Location Location;

    private String basePath;


    public String getBasePath() {
        return basePath;
    }

    public void setBasePath(String basePath) {
        this.basePath = String.format("%s%s", basePath, FileSystems.getDefault().getSeparator());
    }

    public String getRelativePath() {
        return Location.Filename.replace(basePath, "");
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/model/Finding.kt
package com.aquasecurity.plugins.trivy.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.nio.file.FileSystems

class Finding {
    @JsonProperty("rule_id")
    var RuleID: String? = null

    @JsonProperty("long_id")
    var LongID: String? = null

    @JsonProperty("rule_description")
    var RuleDescription: String? = null

    @JsonProperty("rule_provider")
    var Provider: String? = null

    @JsonProperty("rule_service")
    var Service: String? = null

    @JsonProperty("impact")
    var Impact: String? = null

    @JsonProperty("resolution")
    var Resolution: String? = null

    @JsonProperty("links")
    var Links: List<String>? = null

    @JsonProperty("description")
    var Description: String? = null

    @JsonProperty("severity")
    var Severity: Severity? = null

    @JsonProperty("status")
    var Status: Int = 0

    @JsonProperty("resource")
    var Resource: String? = null

    @JsonProperty("location")
    var Location: Location? = null

    private var basePath: String? = null


    fun getBasePath(): String? {
        return basePath
    }

    fun setBasePath(basePath: String?) {
        this.basePath = String.format("%s%s", basePath, FileSystems.getDefault().separator)
    }

    val relativePath: String
        get() = Location!!.Filename.replace(basePath!!, "")
}